<a href="https://colab.research.google.com/github/Jyothif/projects-using-python-and-ML/blob/master/Fake_News_Classifier.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
import pandas as pd
import numpy as np
import itertools
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import PassiveAggressiveClassifier
from sklearn.metrics import accuracy_score,confusion_matrix


In [30]:
#read the data
df = pd.read_csv("/content/drive/MyDrive/Datasets/news.csv")

print("-"*75)
print("The shape of the data set:",df.shape)
print("-"*75)
print("Head of the dataset:\n",df.head())



---------------------------------------------------------------------------
The shape of the data set: (6335, 4)
---------------------------------------------------------------------------
Head of the dataset:
    Unnamed: 0  ... label
0        8476  ...  FAKE
1       10294  ...  FAKE
2        3608  ...  REAL
3       10142  ...  FAKE
4         875  ...  REAL

[5 rows x 4 columns]


In [29]:
labels = df.label
print("-"*75)
print("printing the head of the labels:\n",labels.head())




---------------------------------------------------------------------------
printing the head of the labels:
 0    FAKE
1    FAKE
2    REAL
3    FAKE
4    REAL
Name: label, dtype: object


In [8]:
#splitting the train and test data

X_train,X_test,y_train,y_test = train_test_split(df['text'],
                                                 labels,test_size=0.2,random_state=7)

In [10]:
tfidf_vect = TfidfVectorizer(stop_words='english',max_df=0.7)


tfidf_train= tfidf_vect.fit_transform(X_train)
tfidf_test = tfidf_vect.transform(X_test)

In [13]:
pac = PassiveAggressiveClassifier(max_iter=50)
pac.fit(tfidf_train,y_train)


y_pred = pac.predict(tfidf_test)
score = accuracy_score(y_test,y_pred)
print(f'accuracy:{round(score*100,2)}%')

accuracy:92.82%


In [14]:
confusion_matrix(y_test,y_pred,labels = ['FAKE','REAL'])

array([[590,  48],
       [ 43, 586]])

WE HAVE 
- 590 TRUE POSITIVES,
- 586 TRUE NEGATIVES,
- 43 FALSE POSTIVES
- 48 FALSE NEGATIVES